src/createServer.js

```diff
@@ -1,3 +1,58 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const CASES = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const textForConverting = req.url.split('?')[0].slice(1);
+    const params = Object.fromEntries(normalizedURL.searchParams.entries());
+    const errors = [];
+
+    if (textForConverting.length === 0) {
+      errors.push({
+        message: 'Text to convert is required. '
+          + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!params.toCase) {
+      errors.push({
+        message: '"toCase" query param is required. '
+          + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (!CASES.includes(params.toCase)) {
+      errors.push({
+        message: 'This case is not supported. '
+          + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+
+      res.end(
+        JSON.stringify({ errors }),
+      );
```

Comment:
This can be chained
```suggestion
      res.writeHead(400, { 'Content-Type': 'application/json' })
        .end(
        JSON.stringify({ errors }),
      );
```